src/process.cr

```diff
@@ -301,7 +301,7 @@ class Process
   private def stdio_to_fd(stdio : Stdio, for dst_io : IO::FileDescriptor) : IO::FileDescriptor
     case stdio
     when IO::FileDescriptor
-      stdio
+      stdio.as(IO::FileDescriptor)
```

Comment:
Couldn't the type be derived from the `when` expression?

Reply:
No, because the variable is closured so it always gets all the types. This is the breaking change I mention in my comments.

Reply:
I'd prefer to introduce a new local variable (assigned the value of `stdio`) only in the branch with the closure.

Reply:
Fixed! Not needed anymore.